### PR TITLE
[PRO-1956] map a package to a list of devices

### PR DESCRIPTION
**Pull request by Danten**
Add an association between package-names and device-uuids that will be used for automatically queue installs when the package is updated (which is JIRA ticket PRO-1957)

### CODE REVIEW

**Comment by Danten** [comment]
I realised that one more end-point is needed so will implement that, so be aware that more code is incoming if you are reviewing.

File: core/src/main/resources/db/migration/V27__add_auto_install.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE `AutoInstall` (
```
Comment (simao) [inline core/src/main/resources/db/migration/V27__add_auto_install.sql]: why do you need a namespace and a pkg_name? Why cant we use a package_uuid ?

Comment (Danten) [inline core/src/main/resources/db/migration/V27__add_auto_install.sql]: We want to automatically install whenever a version of this package is published (e.g every push to treehub) so I don't have a version number and therefore not a package_uuid.

File: core/src/main/scala/org/genivi/sota/core/PackagesResource.scala [hunk 2]
```diff
@@ -40,13 +40,16 @@ import org.genivi.sota.http.Errors.RawError
...
+
   /**
     * A scala HTTP routing directive that extracts a package name/version from
     * the request URL
     */
-  val extractPackageId: Directive1[PackageId] = (
-    refined[PackageId.ValidName](Slash ~ Segment) &
-      refined[PackageId.ValidVersion](Slash ~ Segment)).as(PackageId.apply _)
+  val extractPackageId: Directive1[PackageId] =
+    (extractPackageName & refined[PackageId.ValidVersion](Slash ~ Segment)).as(PackageId.apply _)
```
Comment (simao) [inline core/src/main/scala/org/genivi/sota/core/PackagesResource.scala]: thought we had implicits for this already?

Comment (simao) [inline core/src/main/scala/org/genivi/sota/core/PackagesResource.scala]: pretty sure we were using package ids in the url at some point and we were using these directives?

**Pull request merged**